src/main/java/picard/analysis/TheoreticalSensitivity.java

```diff
@@ -204,4 +205,180 @@ public int draw() {
         }
         return normalizedHistogram;
     }
-}
+
+    /**
+     * Determines if a variant would be called under the particular conditions of a given total depth, alt depth,
+     * average base qualities, allele fraction of variant and log odds threshold necessary to call variant.
+     * @param totalDepth Depth at the site to be called, both alt and ref.
+     * @param altDepth Number of alt bases at this site.
+     * @param sumOfAltQualities Average Phred-scaled quality of bases
+     * @param alleleFraction Allele fraction we are attempting to detect
+     * @param logOddsThreshold Log odds threshold necessary for variant to be called
+     * @return
+     */
+    public static boolean isCalled(final int totalDepth, final int altDepth, final double sumOfAltQualities, final double alleleFraction, final double logOddsThreshold) {
+        final double threshold = 10.0 * (altDepth * -Math.log10(alleleFraction) + (totalDepth - altDepth) * -Math.log10(1.0 - alleleFraction) + logOddsThreshold);
+
+        return sumOfAltQualities > threshold;
+    }
+
+    /**
+     * Draw from a binomial distribution.
+     * @param trials Number of trials to perform
+     * @param p Probability of individual success
+     * @param uniformRNG Random number generator to use for making draw
+     * @return Number of total successes
+     */
+    public static int binomialDraw(final int trials, final double p, final Random uniformRNG) {
+        if (p > 1.0 || p < 0) {
+            throw new PicardException("Probabilities should be between 0 and 1, found value " + p + ".");
+        }
+
+        int successes = 0;
+        for (int i = 0; i < trials; i++) {
+            if (uniformRNG.nextDouble() < p) {
+                successes++;
```

Comment:
Does Picard have Apache Commons as a dependency?  If so, their `BinomialDistribution` sampling method would be much faster (O(1)) than an explicit sum of Bernoulli's, which is of course O(N).